woocommerce/woocommerce-ios
# Login Enhancement: Add Jetpack setup flow to Jetpack error screen

Part of #7269 


### Description

This PR integrates the Jetpack setup flow to the login flow using a web view that links to the Jetpack connect webpage.

Changes:
- Add a new view controller with a web view to load the Jetpack connection page.
- Update `JetpackErrorViewModel` to replace the "See Instruction" button with "Install Jetpack" and present the Jetpack Setup screen instead of showing the Jetpack setup instructions.
- Handle callback when Jetpack setup completes by resyncing the site list and attempting to present the epilogue screen again.

Note:
- The design and copies for this screen are still under review (p1658222035244859-slack-C0354HSNUJH) so it's not final yet and might be updated in a future PR. 
- The URL used for `mobile_redirect` is having `wordpress://` scheme due to a hardcoded value in Calypso, and it should be updated when [the PR fixing this](https://github.com/Automattic/wp-calypso/pull/65715) is merged and deployed. 

### Testing instructions

1. Create a JN site with [Jetpack Beta](https://jetpack.com/download-jetpack-beta/), deactivate and delete Jetpack plugin (but keep Jetpack Beta).
2. On the app, log out if necessary, then select Enter Site Address on the prologue screen.
3. Enter the JN site address, then log in with your WP.com account.
4. Notice that the Jetpack error screen is presented because your site doesn't have Jetpack. 
5. Select the Install Jetpack button and notice that a web view is presented for setting up Jetpack.
6. Follow the instruction on the web page for Jetpack setup. If you reach a page about an invalid client ID (an [issue](https://github.com/Automattic/jetpack/pull/25099) of Jetpack connection), dismiss the web view.
7. Navigate to the wp-admin page of your test site, select Jetpack > Jetpack Beta. Select "Manage" on the Jetpack row, and select the Release Candidate (11.2.xxx).
8. Retry steps 5 and 6. Notice that after authorizing Jetpack with your WP.com account, the web view is dismissed and you'll be navigated to the home screen.

### Screenshots


https://user-images.githubusercontent.com/5533851/179896168-3626102c-df2c-4ac4-b7f8-a5e635f27b39.mp4



---
